File: 백준/Bronze/21965. 드높은 남산 위에 우뚝 선/드높은 남산 위에 우뚝 선.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        Scanner sc = new Scanner(System.in);
        boolean increase = true;
        String output = "YES";
        int N = sc.nextInt();
        int check = 0;
        for (int i = 0; i < N; i++) {
            int A = sc.nextInt();
            if (check == A) {
                output = "NO";
                break;
            }
            if (increase) {
                if (check < A) {
                    check = A;
                } else {
                    increase = false;
                }
            } else {
                if (check > A) {
                    check = A;
                } else {
                    output = "NO";
                    break;
                }
            }
        }
        sc.close();
        System.out.println(output);
    }
}

File: 백준/Bronze/21965. 드높은 남산 위에 우뚝 선/드높은 남산 위에 우뚝 선.py
increase = True
output = "YES"

N = int(input())
A = list(map(int, input().split()))
check = 0
for i in A:
    if check == i:
        output = "NO"
        break
    if increase:
        if check < i:
            check = i
        else:
            increase = False
    else:
        if check > i:
            check = i
        else:
            output = "NO"
            break

print(output)